### PR TITLE
Allow for major version redirects to work for root_controller

### PR DESCRIPTION
**Pull request by syncrou**
This allows for `/v1/openapi.json` to redirect to the latest minor version `v1.1`

Also fixes an issue where `/v1.1/openapi.json` was returning a 400.

Below are examples of the api requests and their subsequent log entries:
```
curl localhost:3000/api/v1.1/openapi.json

[----] I, [2020-03-18T13:39:21.993990 #91634:3fdb98032e98]  INFO -- : Started GET "/api/v1.1/openapi.json" for ::1 at 2020-03-18 13:39:21 -0400
[----] I, [2020-03-18T13:39:22.058501 #91634:3fdb98032e98]  INFO -- : Processing by Api::V1x1::RootController#openapi as */*
[----] I, [2020-03-18T13:39:22.058770 #91634:3fdb98032e98]  INFO -- : Request started http://localhost:3000/api/v1.1/openapi.json
[----] I, [2020-03-18T13:39:22.068380 #91634:3fdb98032e98]  INFO -- : Request ended http://localhost:3000/api/v1.1/openapi.json
[----] I, [2020-03-18T13:39:22.068585 #91634:3fdb98032e98]  INFO -- : Completed 200 OK in 10ms (Views: 5.6ms | ActiveRecord: 0.0ms)

curl localhost:3000/api/v1.0/openapi.json

[----] I, [2020-03-18T13:40:21.579903 #91634:3fdb98032e0c]  INFO -- : Started GET "/api/v1.0/openapi.json" for ::1 at 2020-03-18 13:40:21 -0400
[----] I, [2020-03-18T13:40:21.608075 #91634:3fdb98032e0c]  INFO -- : Processing by Api::V1x0::RootController#openapi as */*
[----] I, [2020-03-18T13:40:21.608311 #91634:3fdb98032e0c]  INFO -- : Request started http://localhost:3000/api/v1.0/openapi.json
[----] I, [2020-03-18T13:40:21.614238 #91634:3fdb98032e0c]  INFO -- : Request ended http://localhost:3000/api/v1.0/openapi.json
[----] I, [2020-03-18T13:40:21.614425 #91634:3fdb98032e0c]  INFO -- : Completed 200 OK in 6ms (Views: 5.7ms | ActiveRecord: 0.0ms)

curl -L localhost:3000/api/v1/openapi.json

[----] I, [2020-03-18T13:41:32.936119 #91634:3fdb98032d80]  INFO -- : Started GET "/api/v1/openapi.json" for ::1 at 2020-03-18 13:41:32 -0400
[----] I, [2020-03-18T13:41:32.954490 #91634:3fdb98032d80]  INFO -- : Started GET "/api/v1.1/openapi.json" for ::1 at 2020-03-18 13:41:32 -0400
[----] I, [2020-03-18T13:41:32.963887 #91634:3fdb98032d80]  INFO -- : Processing by Api::V1x1::RootController#openapi as */*
[----] I, [2020-03-18T13:41:32.964116 #91634:3fdb98032d80]  INFO -- : Request started http://localhost:3000/api/v1.1/openapi.json
[----] I, [2020-03-18T13:41:32.970811 #91634:3fdb98032d80]  INFO -- : Request ended http://localhost:3000/api/v1.1/openapi.json
[----] I, [2020-03-18T13:41:32.971027 #91634:3fdb98032d80]  INFO -- : Completed 200 OK in 7ms (Views: 6.1ms | ActiveRecord: 0.0ms)
```

JIRA: https://projects.engineering.redhat.com/browse/SSP-1318

### CODE REVIEW

**Comment by syncrou** [comment]
@bzwei This is ready for review.

I was going to add a new spec, but found that `spec/controllers/v1/root_controller_spec.rb` is already testing that the redirect is working.

This trouble was due to the controllers not pointing to the right openapi `Doc` version. The routes were doing the right thing.

File: app/controllers/api/v1.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 module Api
   module V1
+    class RootController < ApplicationController
```
Comment (bzwei) [inline app/controllers/api/v1.rb]: This file is identical to `app/controller/api/v1/root_contoller.rb` except for the path. How does it work? Why here  `Api::Doc["1.0"]` and your actual test for `v1/openapi.json` returns 1.1?

Comment (syncrou) [reply app/controllers/api/v1.rb]: If `v1` is passed in as a route it never gets to this file. The redirect here calls the latest minor version listed here: https://github.com/RedHatInsights/approval-api/blob/master/config/routes.rb#L13

Comment (syncrou) [reply app/controllers/api/v1.rb]: Though to your point. This file will go away in a followup PR so that should also clear up the redundancy.

**Comment by mkanoor** [comment]
@syncrou even the GraphQL controller is not working
https://ci.cloud.redhat.com/api/approval/v1.1/graphql

**Comment by syncrou** [comment]
@mkanoor I see why. `graphql_controller` isn't included in the inheritance for v1.1.  I'll open another PR to fix that.

**Pull request merged**